extensions/aris/src/bigDataCluster/commands.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as nls from 'vscode-nls';
+import { AppContext } from '../appContext';
+import { TreeNode } from './tree/treeNode';
+import { AddControllerDialog } from './dialog/addControllerDialog';
+import { ControllerTreeDataProvider } from './tree/controllerTreeDataProvider';
+import { ControllerNode, ControllerTreeNode } from './tree/controllerTreeNode';
+import { IEndPoint } from './controller/types';
+
+const localize = nls.loadMessageBundle();
+
+export function registerCommands(appContext: AppContext, treeDataProvider: ControllerTreeDataProvider): void {
+	addBdcController(appContext, treeDataProvider);
+	deleteBdcController(appContext, treeDataProvider);
+	refresh(appContext, treeDataProvider);
+}
+
+function addBdcController(appContext: AppContext, treeDataProvider: ControllerTreeDataProvider): void {
+	appContext.apiWrapper.registerCommand('bigDataClusters.command.addController', (node: TreeNode) => {
```

Comment:
Writing things this ways means it's not possible to unit test:
1. Please leave the `registerCommand` logic in the main extension.ts.
2. Please define the functions that accept a TreeNode and perform the action here. That way, you can unit test the function without having to call the command. The current implementation isn't testable in any way
3. Write unit tests 😄

Reply:
1. Fixed.
2. Fixed.
3. Will add following PR.